src/routing/core.js

```diff
@@ -52,7 +54,10 @@ export function setup(app) {
   app.use([coreBuildRoutes, "/narratives/*"], setSource(req => new CoreSource())); // eslint-disable-line no-unused-vars
 
   app.routeAsync(coreBuildRoutes)
-    .all(setDataset(req => req.path), canonicalizeDataset(path => `/${path}`))
+    .all(
+      setDataset(req => req.path),
+      canonicalizeDataset((req, path) => `/${path}` + (Object.keys(req.query).length ? `?${querystring.stringify(req.query)}` : ''))
```

Comment:
Separately, the use of `req.query` here with `querystring.stringify` brings up an asymmetry we should fix.¹ `req.query` is produced by parsing the request URL's query string with [`qs`](https://www.npmjs.com/package/qs) since don't set Express' `query parser` option. The parsing behaviour of `qs` is a bit different from `querystring` and leads to not roundtripping, e.g.

```console
> qs.parse("a[b]=c")
{ a: { b: 'c' } }
> querystring.stringify(qs.parse("a[b]=c"))
'a='
```

The best solution is not roundtripping, but if we're going to roundtrip, we should be consistent. As `qs` has behaviour I don't think we want nor expect, I think best choice here is to add

    app.set("query parser", "simple"); // querystring.parse()

to our other app settings (like `json replacer`). This would let us roundtrip `req.query` with `querystring.stringify()` and thus also `url.format()` with peace of mind.

See the [`req.query` docs](https://expressjs.com/en/api.html#req.query) and the [`query parser` setting docs](https://expressjs.com/en/api.html#app.settings.table).

¹ This asymmetry was sort of pre-existing, with the use of `url.format({…, query: req.query})` in one spot, which you've also recently touched in #791.

Reply:
Switched to parsing with `querystring` in [847282a](https://github.com/nextstrain/nextstrain.org/pull/793/commits/847282a19b386b82d7e8cb1532f4809caa32f680)